gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripEdgeArrivalDepartureTimeValidatorTest.java

```diff
@@ -0,0 +1,333 @@
+                                                   .setStopSequence(2)
+                                                   .build();
+  private final GtfsStopTime middleStopTrip2 = new GtfsStopTime.Builder()
+                                                   .setCsvRowNumber(41)
+                                                   .setTripId("trip2")
+                                                   .setStopId("1002")
+                                                   .setStopSequence(2)
+                                                   .build();
+  private final GtfsStopTime endStopTimeHasNeitherTrip2 = new GtfsStopTime.Builder()
+                                                              .setCsvRowNumber(25)
+                                                              .setTripId("trip2")
+                                                              .setStopId("1003")
+                                                              .setStopSequence(3)
+                                                              .build();
+  private final GtfsStopTime startStopTimeHasNeitherTrip2 = new GtfsStopTime.Builder()
+                                                                .setCsvRowNumber(20)
+                                                                .setTripId("trip2")
+                                                                .setStopId("1001")
+                                                                .setStopSequence(1)
+                                                                .build();
+  @Test
+  public void validShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    TripEdgeArrivalDepartureTimeValidator validator = new TripEdgeArrivalDepartureTimeValidator();
+
+    // Create stopTimeTable:
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(startStopTimeHasBothTrip1);
+    stopTimes.add(middleStopTrip1);
+    stopTimes.add(endStopTimeHasBothTrip1);
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+    // Create tripTable:
+    List<GtfsTrip> trips = new ArrayList<>();
+    trips.add(trip1);
+    validator.tripTable = GtfsTripTableContainer.forEntities(trips, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+
+  @Test
+  public void missingDepartureTimeForFirstStop() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    TripEdgeArrivalDepartureTimeValidator validator = new TripEdgeArrivalDepartureTimeValidator();
+
+    // Create stopTimeTable:
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(startStopTimeHasArrivalTrip1);
+    stopTimes.add(middleStopTrip1);
+    stopTimes.add(endStopTimeHasBothTrip1);
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
```

Comment:
Could this code be simplified and more readable if you used something like:
validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(List.of(startStopTimeHasArrivalTrip1, middleStopTrip1, endStopTimeHasBothTrip1), noticeContainer);

Reply:
ok using Arrays.asList as List.of is only available with Java9